lib/src/common/enums.dart

```diff
@@ -0,0 +1,430 @@
+part of tiled;
```

Comment:
sadly we have to have a bit of boilerplate for enums because of dart lacks.
I might consider breaking this off into multiple files later, but for now this is good.

Reply:
I have a vague memory that the use of "part of" was no longer recommended, but I can't find anything about it in the style guide. Does anyone else remember this?

Reply:
@renancaraujo

Reply:
discussed this with @spydon , we will address on a followup!